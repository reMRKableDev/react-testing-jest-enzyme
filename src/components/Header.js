import React from "react";

const Header = ({ name }) => {
  return name ? <h1>Hello {name}!</h1> : <h1>Hello stranger!</h1>;
};

export default Header;
