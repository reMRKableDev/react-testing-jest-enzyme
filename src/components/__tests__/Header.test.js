import React from "react";
import Header from "../Header";
import { shallow, render, mount } from "enzyme";

/* GLOBAL VARIABLES, MATCHERS & FUNCTIONS */
const HeaderComponent = <Header />;
const HeaderComponentWithNameProp = <Header name="Malcolm" />;

const expectMatchSnapshot = wrapper => expect(wrapper).toMatchSnapshot();
const expectToEqual = (receivedValue, expectedValue) =>
  expect(receivedValue).toEqual(expectedValue);
const findElementValue = wrapper => wrapper.find("h1").text();

/* TESTS */
test("shallow rendering component correctly", () => {
  const wrapper = shallow(HeaderComponent);
  expectMatchSnapshot(wrapper);
});

test("shallow rendering correctly component with name", () => {
  const wrapper = shallow(HeaderComponentWithNameProp);
  expectMatchSnapshot(wrapper);
});

test("rendering component", () => {
  const wrapper = render(HeaderComponent);
  expectMatchSnapshot(wrapper);
});

test("rendering component with name", () => {
  const wrapper = render(HeaderComponentWithNameProp);
  expectMatchSnapshot(wrapper);
});

test("mounting component", () => {
  const wrapper = mount(HeaderComponent);
  const receivedTextValue = findElementValue(wrapper);
  expectToEqual(receivedTextValue, "Hello stranger!");
});

test("mounting component with name", () => {
  const wrapper = mount(HeaderComponentWithNameProp);
  const receivedTextValue = findElementValue(wrapper);
  expectToEqual(receivedTextValue, "Hello Malcolm!");
});
