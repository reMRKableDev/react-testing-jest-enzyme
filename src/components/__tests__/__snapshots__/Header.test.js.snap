// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering component 1`] = `
<h1>
  Hello stranger!
</h1>
`;

exports[`rendering component with name 1`] = `
<h1>
  Hello Malcolm!
</h1>
`;

exports[`shallow rendering component correctly 1`] = `
<h1>
  Hello stranger!
</h1>
`;

exports[`shallow rendering correctly component with name 1`] = `
<h1>
  Hello 
  Malcolm
  !
</h1>
`;
